### ISSUE TITLE
Error with Install Snap 3.11.3 on Moodle 3.11.5

### ISSUE THREAD

**Issue by peter-lernyx** [open-lms-open-source/moodle-theme_snap#228]
![image](https://user-images.githubusercontent.com/75390480/153762464-77aadb61-cfba-4aac-b548-1dc82bf8cc51.png)

**Comment by peter-lernyx** [comment]
Only remedy for now seems to be to completely remove Snap from the install.

**Comment by tlock** [comment]
On which version of PHP and stack @peter-lernyx ?

**Comment by peter-lernyx** [comment]
Good point it's 7.3.33 and it's flagged as outdated. Should I try and update maybe?

**Comment by peter-lernyx** [comment]
Nope after PHP upgrade to 7.4.27 template installs but creates complete mayhem.

New debug messages:
get_user_field_name() is deprecated. Please use \core_user\fields::get_display_name() instead
line 3366 of /lib/deprecatedlib.php: call to debugging()
line 42 of /theme/snap/settings/profile_based_branding.php: call to get_user_field_name()
line 41 of /theme/snap/settings.php: call to require()
line 57 of /admin/settings/appearance.php: call to include()
line 8855 of /lib/adminlib.php: call to require()
line 40 of /admin/category.php: call to admin_get_root()
get_user_field_name() is deprecated. Please use \core_user\fields::get_display_name() instead
line 3366 of /lib/deprecatedlib.php: call to debugging()
line 43 of /theme/snap/settings/profile_based_branding.php: call to get_user_field_name()
line 41 of /theme/snap/settings.php: call to require()
line 57 of /admin/settings/appearance.php: call to include()
line 8855 of /lib/adminlib.php: call to require()
line 40 of /admin/category.php: call to admin_get_root()
get_user_field_name() is deprecated. Please use \core_user\fields::get_display_name() instead
line 3366 of /lib/deprecatedlib.php: call to debugging()
line 44 of /theme/snap/settings/profile_based_branding.php: call to get_user_field_name()
line 41 of /theme/snap/settings.php: call to require()
line 57 of /admin/settings/appearance.php: call to include()
line 8855 of /lib/adminlib.php: call to require()
line 40 of /admin/category.php: call to admin_get_root()
get_user_field_name() is deprecated. Please use \core_user\fields::get_display_name() instead
line 3366 of /lib/deprecatedlib.php: call to debugging()
line 45 of /theme/snap/settings/profile_based_branding.php: call to get_user_field_name()
line 41 of /theme/snap/settings.php: call to require()
line 57 of /admin/settings/appearance.php: call to include()
line 8855 of /lib/adminlib.php: call to require()
line 40 of /admin/category.php: call to admin_get_root()
get_user_field_name() is deprecated. Please use \core_user\fields::get_display_name() instead
line 3366 of /lib/deprecatedlib.php: call to debugging()
line 46 of /theme/snap/settings/profile_based_branding.php: call to get_user_field_name()
line 41 of /theme/snap/settings.php: call to require()
line 57 of /admin/settings/appearance.php: call to include()
line 8855 of /lib/adminlib.php: call to require()
line 40 of /admin/category.php: call to admin_get_root()
String [grade,core] is deprecated. Either you should no longer be using that string, or the string has been incorrectly deprecated, in which case you should report this as a bug. Please refer to https://docs.moodle.org/dev/String_deprecation
line 394 of /lib/classes/string_manager_standard.php: call to debugging()
line 7278 of /lib/moodlelib.php: call to core_string_manager_standard->get_string()
line 155 of /mod/lesson/settings.php: call to get_string()
line 89 of /lib/classes/plugininfo/mod.php: call to include()
line 47 of /admin/settings/plugins.php: call to core\plugininfo\mod->load_settings()
line 8857 of /lib/adminlib.php: call to require()
line 40 of /admin/category.php: call to admin_get_root()

**Comment by tlock** [comment]
Those warnings are related to Moodle 3.11.x not PHP, strange that an out of date PHP 7.3.x would cause the issue in the first place.

**Comment by peter-lernyx** [comment]
I have it working now. When I installed from within Moodle, strangely it pulled an old version of Snap as per the plugin overview.

![image](https://user-images.githubusercontent.com/75390480/153776774-40aa2059-e858-4a25-86ba-ab19cf4addc7.png)

After uninstalling, manually copying the 3.11.3 version directly to the server and upgrading the database, it now works fine and has the right version installed. There still is a debugging message though:

tring [grade,core] is deprecated. Either you should no longer be using that string, or the string has been incorrectly deprecated, in which case you should report this as a bug. Please refer to https://docs.moodle.org/dev/String_deprecation
line 394 of /lib/classes/string_manager_standard.php: call to debugging()
line 7278 of /lib/moodlelib.php: call to core_string_manager_standard->get_string()
line 155 of /mod/lesson/settings.php: call to get_string()
line 89 of /lib/classes/plugininfo/mod.php: call to include()
line 47 of /admin/settings/plugins.php: call to core\plugininfo\mod->load_settings()
line 8857 of /lib/adminlib.php: call to require()
line 511 of /lib/classes/plugininfo/base.php: call to admin_get_root()
line 1805 of /admin/renderer.php: call to core\plugininfo\base->get_settings_url()
line 352 of /admin/renderer.php: call to core_admin_renderer->plugins_control_panel()
line 215 of /admin/plugins.php: call to core_admin_renderer->plugin_management_page()

**Comment by tlock** [comment]
Hi Peter,
That is strange because the 3.11.3 version has been uploaded to Moodle Plugins over 7 days ago.

Error you are seeing is from the Lesson module:

line 155 of /mod/lesson/settings.php: call to get_string()

**Comment by peter-lernyx** [comment]
Oh, excellent, will investigate and report a bug on Moodle. Thanks a lot! ;-)

**Comment by peter-lernyx** [comment]
Had the same thing again now and think I know what is going on. When you install from inside moodle through the pluging directory link, it downloads and installs this old version somehow. How that gets across the version check I do not know but that's what I did on 3.11.5+ and this is what I got:
![grafik](https://user-images.githubusercontent.com/75390480/156223091-5be6311a-06ee-47c6-9cd6-74f8a8759e3a.png)

**Comment by peter-lernyx** [comment]
Oh well. Did an install, downloaded the snap file manually from the plugin directory and performed a drag and drop install. All checks pass ok, but upon upgrading the database, now this (on 3.11.5+):
![grafik](https://user-images.githubusercontent.com/75390480/156223748-04686ec5-79b3-46be-be64-f4153c12e7eb.png)

**Comment by xKoetsuji** [comment]
Hello Peter, can you please tell me which PHP version are you using? This must to be fixed just upgrading to PHP 7.4

**Comment by peter-lernyx** [comment]
I thought I had done this before when this happend last. Let me check

**Comment by peter-lernyx** [comment]
the hell??
![grafik](https://user-images.githubusercontent.com/75390480/156224498-a9deb122-2a2a-49e0-962c-6aaa3965776e.png)

**Comment by xKoetsuji** [comment]
Yeah... please try again using PHP 7.4

**Comment by peter-lernyx** [comment]
waddayaknow...
It works again. I am positive that I changed this because everything worked just find before the upgrade to 3.11.5+. have a word with the server admin I gues.. 
Thank you very much for fixing the same error twice... ;-)
